pyscf/mcscf/ucasci.py

```diff
@@ -404,6 +404,21 @@ def make_rdm1(self, mo_coeff=None, ci=None, ncas=None, nelecas=None,
         dm1a,dm1b = self.make_rdm1s(mo_coeff, ci, ncas, nelecas, ncore)
         return dm1a+dm1b
 
+    def fix_spin_(self, shift=.2, ss=None):
+        r'''Use level shift to control FCI solver spin.
+
+        .. math::
+
+            (H + shift*S^2) |\Psi\rangle = E |\Psi\rangle
+
+        Kwargs:
+            shift : float
+                Level shift for states which have different spin
+            ss : number
+                S^2 expection value == s*(s+1)
+        '''
+        fci.addons.fix_spin_(self.fcisolver, shift, ss)
```

Comment:
This function works for restricted orbitals only. The integrals of operators S_x, S_y have to be explicitly considered for unrestricted orbitals. They are not simply zero or one as the spatial degenerated orbitals. If alpha and beta orbitals are very different, calling the restricted version would cause large error.